### PR TITLE
Enable ethernet on RW612

### PR DESCRIPTION
**Pull request by decsny**
Enable ethernet on RW612 SOC and RD_RW612_BGA board as well as a kconfig fix. This PR adds a board variant to build for ethernet because some board hardware rework is required to use the ethernet which will conflict with some other functions of the board.

### CODE REVIEW

File: drivers/ethernet/nxp_enet/eth_nxp_enet.c [hunk 1]
```diff
@@ -11,6 +11,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include "zephyr/sys_clock.h"
```
Comment (danieldegrasse) [inline drivers/ethernet/nxp_enet/eth_nxp_enet.c:14]: Unclear here- why are we adding the `sys_clock.h` include here? Was it always needed, and this PR just caught the requirement for the include?

Comment (decsny) [reply drivers/ethernet/nxp_enet/eth_nxp_enet.c:14]: this is quite a mystery, my suspect lineup involves a set of various dubious mouse "controls" that tend to serve no purpose except annoy me

File: boards/nxp/rd_rw612_bga/doc/index.rst [hunk 2]
```diff
@@ -167,8 +167,36 @@ In this case, the BLE blob won't be linked with the application, so the user nee
...
 
+Board variants
+**************
+
+Ethernet
+========
+
+To use ethernet on the RD_RW612_BGA board, you first need to make the following
+modifications to the board hardware:
+
```
Comment (butok) [inline boards/nxp/rd_rw612_bga/doc/index.rst:178]: Will this be fixed in the next version of the RW612 board?

Comment (decsny) [reply boards/nxp/rd_rw612_bga/doc/index.rst:178]: I have no idea what NXP plans are for future versions of the RD board, but on the FRDM board, it doesn't require any modification, and I'm gonna make a PR for that next

**Comment by decsny** [comment]
rebased, removed stray line, added commit for frdm_rw612, and updated properties of ksz8081 phy for upstream name change

**Review by danieldegrasse — DISMISSED** [review]

File: boards/nxp/rd_rw612_bga/doc/index.rst (hunk 2 above)
Comment (decsny) [reply boards/nxp/rd_rw612_bga/doc/index.rst:178]: added a commit for frdm rw612

**Review by pdgendt — DISMISSED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**